tools/OpenConsole.psm1

```diff
@@ -7,16 +7,36 @@ Set-Item -force -path "env:OpenConsoleRoot" -value "$PSScriptRoot\.."
 # them into the Powershell environment.
 function Set-MsbuildDevEnvironment()
 {
-    $path = "$env:VS140COMNTOOLS\..\.."
-    pushd $path
-    cmd /c "vcvarsall.bat&set" | foreach {
-        if ($_ -match "=")
+    $vswhere = ${env:ProgramFiles(x86)} + '\Microsoft Visual Studio\Installer\vswhere.exe'

```

Comment:
I'd prefer if we go with the solution we settled on for `razzle.cmd`. We have a nuget package list in the root of the repository that expresses a dependency on vswhere so that it gets installed locally.

Alternatively, there's [this](https://github.com/microsoft/vssetup.powershell), the VSSetup PowerShell module. It's a powershell interface to the same API that VSWhere uses! It seems very cool.

Reply:
Like vswhere, that module is _also_ by @heaths!

Reply:
unfortunately, I don't know how one would do this (not a Powershell person!), but would that look something like:

```pwsh
  Install-Module VSSetup -Scope CurrentUser
  $vsinfo = Get-VSSetupInstance  -All | `
    Select-VSSetupInstance -Latest -Require 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64' -Product *
  # do the things with vcvarsall
```

Reply:
Correct. https://github.com/microsoft/vswhere/wiki/Start-Developer-Command-Prompt has an example, though using vswhere to find an instance, which you've done there similarly along with a suitable component. IIRC, vsdevcmd.bat always gets installed, but other components light up support for their related features.